src/Console/Command/Compile.php

```diff
@@ -365,15 +380,41 @@ private function registerMainScript(Configuration $config, Box $box, BuildLogger
         return $localMain;
     }
 
-    private function registerStub(Configuration $config, Box $box, string $main, BuildLogger $logger): void
+    private function registerRequirementsChecker(Configuration $config, Box $box, BuildLogger $logger): bool
```

Comment:
add a test in `CompilerTest`; simple however just to double check the files added since there is already and e2e test for it